src/qtcore/qml/elements/Component.js

```diff
@@ -7,6 +7,20 @@ function QMLContext() {
     }
 }
 
+QMLComponent.getAttachedObject = function() { // static
+    if (!this.Component) {
```

Comment:
yes, `this.$Component` is needed to store the actual attached object, `this.Component` is the required public API and is just a "link" to `Component.getAttachedObject()`. I'm fine though with any other way to store the actual attached object. It's completely internal and I'm fine if it's only accessible via getAttachedObject (then we need some hasAttachedObject or get-but-don't-create-if-non-existant kind of function)